### ISSUE TITLE
Bug: fix error message in `no-loop-func`

### ISSUE THREAD

**Issue by snitin315** [eslint/eslint#19586]
### Environment

playground

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint no-loop-func: "error"*/

for (var i = 0; i < 10; i++) {
  items.push({
    id: i,
    name: "Item " + i
  });

  const process = function (callback){
    callback({ id: i, name: "Item " + i });
  };
}
```

### What did you expect to happen?

Error message to mention "i" only once.

`Function declared in a loop contains unsafe references to variable(s) 'i'. `

### What actually happened?

It reported i multiple times in the message.

`Function declared in a loop contains unsafe references to variable(s) 'i', 'i'. `

### Link to Minimal Reproducible Example

[Playground Demo ](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tbG9vcC1mdW5jOiBcImVycm9yXCIqL1xuXG5mb3IgKHZhciBpID0gMDsgaSA8IDEwOyBpKyspIHtcbiAgaXRlbXMucHVzaCh7XG4gICAgaWQ6IGksXG4gICAgbmFtZTogXCJJdGVtIFwiICsgaVxuICB9KTtcblxuICBjb25zdCBwcm9jZXNzID0gZnVuY3Rpb24gKGNhbGxiYWNrKXtcbiAgICBjYWxsYmFjayh7IGlkOiBpLCBuYW1lOiBcIkl0ZW0gXCIgKyBpIH0pO1xuICB9O1xufSIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

### Participation

- [x] I am willing to submit a pull request for this issue.
